### DISCUSSION TITLE
Passing UUID to Filtering Request (Read Data)

### DISCUSSION THREAD

**Discussion by rcsilva81** [supabase/supabase#1156] [Questions] [answered]
Guys, maybe I'm already sleeping. But have been struggling to get the lines below to work. 

`With xmlHttp
    .Open "GET", supabaseAPI & "users?id=eq.'" & uuid & "'select=flag_admin"
    .setRequestHeader "apikey", supabaseAnnonKey
    .setRequestHeader "Authorization: Bearer", supabaseAnnonKey
    .send`

The url bein passed: 
`https://xxxxxxxxxxxxxx.supabase.co/rest/v1/users?id=eq.7edf4090-5856-4b50-a17e-769e81d542a7select=*`

The error I'm receiving: 

`{"hint":null,"details":null,"code":"22P02","message":"invalid input syntax for type uuid: \"'7edf4090-5856-4b50-a17e-769e81d542a7'select=flag_admin\""}`

Can anyone shed some light into what the issue might be?

Tks,
Rob

**Comment by steve-chavez** [comment, marked as answer]
There's a syntax error, query parameters must be separated by `&`, like this:

```
https://xxxxxxxxxxxxxx.supabase.co/rest/v1/users?id=eq.7edf4090-5856-4b50-a17e-769e81d542a7&select=*
```

Otherwise the `<uuid>+select` will be passed as the filter value, which is an invalid uuid as the postgres error describes.

**Reply by rcsilva81** [reply]
Tks Steve, that fixed the error. However, I may be doing something else wrong, cause I'm receiving nothing back. 

![image](https://user-images.githubusercontent.com/68916995/114867202-dfe1bd80-9dca-11eb-8d8a-37ca40ebcf83.png)

But if I ran the query on the database the data is there.
![image](https://user-images.githubusercontent.com/68916995/114867469-3c44dd00-9dcb-11eb-99c1-d0e744e558e3.png)

**Reply by steve-chavez** [reply]
Do you have RLS enabled for that table?

**Reply by rcsilva81** [reply]
Not for that table, but the uuid on this table, references the Auth users uuid column. Would that count?

**Reply by steve-chavez** [reply]
Try using the `service_role`(bypasses RLs) API key to be certain. You can get it on `/settings/api`, then use the jwt in the `Authorization` header.

**Reply by rcsilva81** [reply]
Tks Steve.... using the service_role worked... on the followin format :

`apikey: service_role`
`Authorization: Bearear service_role`

Thanks for the help.
